packages/provider/src/json-rpc-provider.ts

```diff
@@ -35,6 +35,28 @@ import {
 
 const DEFAULT_TIMEOUT = 10000
 
+/**
+ * ExtractBasicAuth extracts basic authentication credentials from the given
+ * url into a separate base64-encoded string so that we can pass it to Fetch API.
+ **/
+export function ExtractBasicAuth(urlStr: string) {
```

Comment:
Nit: why upper camel case? Makes me think it's a react Component/HoC

Reply:
`extractBasicAuth`

Reply:
this was my muscle memory.. thank you for pointing this out!